backend/src/main/java/gov/cdc/usds/simplereport/api/CurrentTenantDataAccessContextHolder.java

```diff
@@ -0,0 +1,41 @@
+package gov.cdc.usds.simplereport.api;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.context.WebApplicationContext;
+
+@Repository
+@Scope(scopeName = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
+public class CurrentTenantDataAccessContextHolder {
+
+  private String username = "";
+  private Set<String> authorityNames = new HashSet<>();
+  private boolean hasBeenPopulated = false;
+
+  public void setTenantDataAccessAuthorities(String username, Set<String> authorityNames) {
+    this.username = username;
```

Comment:
why do you need username? isn't this only called on the current user, and scoped to the current request?

Reply:
The username is used in the Okta repository's `getOrganizationRoleClaimsForUser` so the claims can be overridden based on the tenant data access authorities when the requester is getting their own claims.  Maybe there is a better way to do this?  I couldn't think of anything obvious, though.